### PR TITLE
Reference: type and context definitions to attr_reader

### PR DESCRIPTION
**Pull request by fabianfrz**

### CODE REVIEW

File: lib/opentracing/reference.rb [hunk 1]
```diff
@@ -63,9 +63,7 @@ def self.child_of(context)
     #   ])
     #
     def self.follows_from(context)
-      if context.is_a?(Span)
-        context = context.context
-      end
+      context = context.context if context.is_a?(Span)
```
Comment (iaintshine) [inline lib/opentracing/reference.rb]: Why did u change the line in the PR, seems to be unrelated ? Notice that `self.child_of` contains exactly the same logic for SpanContext extraction.

File: lib/opentracing/reference.rb [hunk 2]
```diff
@@ -76,13 +74,9 @@ def initialize(type, context)
     end
 
     # Return [String] reference type
-    def type
-      @type
-    end
+    attr_reader :type
```
Comment (iaintshine) [inline lib/opentracing/reference.rb]: Can u please change the documentation line above, to be consistent with [YARD's documentation style for documenting attributes](https://github.com/lsegal/yard/blob/master/docs/GettingStarted.md#documenting-attributes).  Use `@return` instead of `Return`.

**Comment by fabianfrz** [comment]
@iaintshine I have not changed anything here to functionality or how it looks. API is still the same but the methods are generated how it should be done in ruby.

An if which has a single command in the block should go to the same line if the line does not become too long.

`attr_reader` creates a method which looks exactly like the one you have written. `attr_writer` creates a setter and `attr_accessor` creates both at once.

**Comment by fabianfrz** [comment]
Also the brackets may be removed but I'll keep them for now

File: lib/opentracing/reference.rb [hunk 3]
```diff
@@ -24,10 +24,7 @@ class Reference
     #   ])
     #
     def self.child_of(context)
-      if context.is_a?(Span)
-        context = context.context
-      end
-
+      context = context.context if context.is_a?(Span)
```
Comment (itsderek23) [inline lib/opentracing/reference.rb]: Couldn't this method just be one line:

```ruby
Reference.new(CHILD_OF, context.try(:context))
```

File: lib/opentracing/reference.rb [hunk 4]
```diff
@@ -75,14 +69,10 @@ def initialize(type, context)
       @context = context
     end
 
-    # Return [String] reference type
-    def type
-      @type
-    end
+    # @return [String] reference type
+    attr_reader :type
```
Comment (itsderek23) [inline lib/opentracing/reference.rb]: 👍

File: lib/opentracing/reference.rb (hunk 3 above)
Comment (indrekj) [reply lib/opentracing/reference.rb]: No. `try` is rails specific method. And even then I think it returns `nil` if there's no such method.

Comment (itsderek23) [reply lib/opentracing/reference.rb]: image

**Comment by itsderek23** [comment]
Included in v0.3.2.
